numpy/core/_asarray.py

```diff
@@ -115,7 +118,7 @@ def require(a, dtype=None, requirements=None, *, like=None):
         subok = True
 
     order = 'A'
-    if requirements >= {'C', 'F'}:
+    if requirements.issuperset(('C', 'F')):
```

Comment:
```suggestion
    if requirements >= {'C', 'F'}:
```
I don't buy it until proven.  And in this case my test indicates the **original** is faster. You are optimizing one thing and forgetting that the other path needs to get the attribute?

Would be nice to proof that the rest is worthwhile as well.  But it seems OK and pushing the definition out is fine in this truly private file.  So maybe its not important enough here.  It seems like pulling out `POSSIBLE_FLAGS` gives around ~20% speed for cases that do not need to copy.

Reply:
Sure thing. Testing these changes separately on 3.10, I do see the same slowdown you do.

The results here vary based on the Python version, the idea was to save on building the two-element set, and benefit from `issuperset` getting a [new path for iterables](https://github.com/python/cpython/commit/a69a4a917c436579c2c4112081ea86a70f1f05d3) in Python 3.11 along with the attribute lookup improvements. It at least doesn't seem to hurt on 3.11, but I hadn't realized the hit is so substantial in earlier Pythons.

Given that it's only marginally helpful at best, and the existing set comparison is more natural anyway, I'll put it back. Good catch, and thanks for taking a look.

<details><summary>Test outputs</summary>

These are all a quick test with `timeit` on a 128x128 array of float64, with three `require` versions: the "original" which builds both the dictionary and the set, the "dict" which hoists the dictionary out, and "both" which then adds the `issuperset` change on top of that. These are versions 3.10.5 and 3.11.0b3.

The first test is a call with an empty requirements list:
```
py310 original: 0.41715210700022
py310 dict:     0.12086745600026916
py310 both:     0.12723797199942055

py311 original: 0.3896337879996281
py311 dict:     0.10250333399926603
py311 both:     0.10182175100089808
```
for that one we never reach the set check so "dict" vs. "both" is just timing noise (just there for completeness).

The second gives requirements `['C', 'A']` which are already satisfied (avoids the copy, which is far more expensive than any of this):
```
py310 original: 1.0781186469994282
py310 dict:     0.7068047739994654
py310 both:     0.7638090609998471

py311 original: 0.986893441000575
py311 dict:     0.6817922239988548
py311 both:     0.6634978880010749
```

</details>